src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/ObjectMetadata.php

```diff
@@ -18,23 +18,40 @@ class ObjectMetadata extends PropertyMetadata
      */
     private $properties;
 
+    /**
+     * @var SchemaMetadata[]
+     */
+    private $anyOfs;
+
+    /**
+     * @var SchemaMetadata[]
+     */
+    private $allOfs;
+
     /**
      * @param PropertyMetadata[] $properties
+     * @param SchemaMetadata[] $anyOfs
+     * @param SchemaMetadata[] $allOfs
      */
-    public function __construct(string $name, bool $mandatory, array $properties = [])
```

Comment:
Are these additions a leftover?

Reply:
No, they are used in the `SingleSelectionPropertyMetadataMapper`

Reply:
Ah, seems like I have overseen that... And I think I was also confused, because the `SingleSelectionProperty` does not return an object, right? Is it correct to use that class then? I would say either the `PropertyMetadata` or the `SchemaMetadata` should be the fallback containing these additional properties, don't you agree?